lib/crystalball/logger.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+module Crystalball
+  # This class is responsible for logging information to the configured output stream.
+  class Logger
+    VALID_LOG_LEVELS = %i[debug info warn error critical].freeze
+    COLOR_CODES = {
+      info: "\e[33m",
+      warn: "\e[91m",
+      error: "\e[31m",
+      critical: "\e[101m"
+    }.freeze
+
+    include Singleton
+
+    class << self
+      extend Forwardable
+
+      delegate %i[
```

Comment:
I think logger should be used as `Crystalball.logger` only, without class level methods
I don't really like an idea to use `Crystalball::Logger` to log anything
what do you think @pluff ?